extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
Upload is not allowed at all if anonymous. Existing checkPermissions() checks that. For Japan, if $wgDisableAnonymousEditing = true, then we can call checkPermissions()  or simply return error. I thought just returning error is a lot cheaper than going down into checkPermissions(). In addition, we are going to add login status to Mercury soon that we have to rewrite the logic here anyways. Having a completely separated function for 'JP' (checkPermissionsForJA()) makes the near-future code change easier, don't you think? We just delete this checkPermissionsForJA() function. 
I keep this as-is.